tests/Sharpliner.Serialization.Tests/MockPipeline.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Sharpliner.Model.AzureDevOps;
+
+namespace Sharpliner.Serialization.Tests
+{
+    internal class MockPipeline : AzureDevOpsPipelineDefinition
+    {
+        public override string TargetFile => "azure-pipelines.yml";
+
+        public override TargetPathType TargetPathType => TargetPathType.RelativeToGitRoot;
+
+        public override Pipeline Pipeline => new()
+        {
+            Name = "$(Date:yyyMMdd).$(Rev:rr)",
+
+            Variables =
+            {
+                new Variable("Configuration", "Release"), // We can create the objects and then resue them for definition too
+                Variable("Configuration", "Release"),     // Or we have this more YAML-like definition
```

Comment:
I dig this version, but I'm sure you knew.

Reply:
I want to keep both so that you can define variables in C# and then just drop them in Jobs/Steps so you can use the benefit of being in the C# program. The same needs to happen for other pieces of the pipeline.

I can imagine that you define your own steps like "Signing Job" as C# objects and then assemble pipelines out of them. The pipeline then could be just 10 lines of objects and super clear to understand